fontforge/parsettf.c

```diff
@@ -3974,18 +3974,20 @@ return( 0 );
     else {
 	fseek(ttf,info->cff_start+dicts[which]->fdarrayoff,SEEK_SET);
 	subdicts = readcfftopdicts(ttf,NULL,info->cff_start,info,dicts[which]);
-	fseek(ttf,info->cff_start+dicts[which]->fdselectoff,SEEK_SET);
-	fdselect = readfdselect(ttf,dicts[which]->glyphs.cnt,info);
-	for ( j=0; subdicts[j]!=NULL; ++j ) {
-	    if ( subdicts[j]->private_offset!=-1 )
-		readcffprivate(ttf,subdicts[j],info);
-	    if ( subdicts[j]->charsetoff!=-1 )
-		readcffset(ttf,subdicts[j],info);
+	if ( subdicts!=NULL ) {
+	    fseek(ttf,info->cff_start+dicts[which]->fdselectoff,SEEK_SET);
+	    fdselect = readfdselect(ttf,dicts[which]->glyphs.cnt,info);
```

Comment:
Are you sure that the statements affecting the read position go inside the conditional?

Reply:
Good question. Of the various `read` functions only one has a `SEEK_CUR` and its in the middle. The general strategy is to start fresh with a `SEEK_SET` or, occasionally, a `SEEK_LAST`. That includes `readcffenc` called below in this function. 

Looking at the question from the other direction, it doesn't seem like past the end of the last subdict for `which`, even given that `which` is (probably?) almost always going to be zero, is enough of a particular place for the code to rely on.